evm_arithmetization/src/cpu/kernel/constants/mod.rs

```diff
@@ -386,8 +391,17 @@ pub mod cancun_constants {
     pub const BEACON_ROOTS_CONTRACT_CODE_HASH: [u8; 32] =
         hex!("f57acd40259872606d76197ef052f3d35588dadf919ee1f0e3cb9b62d3f4b02c");
 
-    pub const BEACON_ROOTS_CONTRACT_ADDRESS_HASHED: [u8; 32] =
-        hex!("37d65eaa92c6bc4c13a5ec45527f0c18ea8932588728769ec7aecfe6d9f32e42");
+    pub const BEACON_ROOTS_CONTRACT_ADDRESS_HASHED: H256 = H256(hex!(
+        "37d65eaa92c6bc4c13a5ec45527f0c18ea8932588728769ec7aecfe6d9f32e42"
+    ));
```

Comment:
@Nashtare you commented that these constants don't belong here.
Should that block this PR, or shall I open an issue to move them?

Reply:
I didn't mean to have it as blocking comment, but given that this is used in both `evm_arithmetization` and `trace_decoder` crates, I believe the address should be defined in the common crate. But can be left to a cleanup PR, it's technically unrelated to the refactor

Reply:
#531